go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -155,6 +156,8 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 			func(tablet *topodatapb.Tablet) error {
 				if tablet.Type != topodatapb.TabletType_MASTER {
 					tablet.Type = topodatapb.TabletType_MASTER
+					// TODO (deepthi): this line causes TestStateChangeImmediateHealthBroadcast to fail
+					//tablet.MasterTermStartTime = logutil.TimeToProto(agent.masterTermStartTime())
```

Comment:
Not sure why this causes the test to fail. Still need to debug and fix it.

Reply:
The failure I'm seeing with this line uncommented is this:

```
healthcheck_test.go:801: unexpected BroadcastData.SecondsBehindMaster, got: 0 want: 20
```

I think it's because we added a call to `setMasterTermStartTime` in `refreshTablet`, if the MasterTermStartTime in the tablet record is set. In turn, `setMasterTermStartTime` sets `agent._replicationDelay = 0`.

I'm not sure why we set the replication delay there, but perhaps the appropriate fix is just to change the test to expect that refreshing a master tablet always resets the replication delay to 0.

Reply:
I know why we set replicationDelay = 0, it's because a master by definition has no lag.
I will change the test.